src/rebar_prv_dialyzer.erl

```diff
@@ -178,7 +181,12 @@ proj_plt_files(State) ->
     PltApps = get_config(State, plt_extra_apps, []),
     Apps = rebar_state:project_apps(State),
     DepApps = lists:flatmap(fun rebar_app_info:applications/1, Apps),
-    get_plt_files(BasePltApps ++ PltApps ++ DepApps, Apps).
+    DepApps1 =
+        case get_config(State, plt_include_all_deps, false) of
```

Comment:
It would be nice not to use a boolean value and instead use an atom describing the strategy. This will allow us flexibility in the future for other options and makes it clearer what the choices are. Perhaps `{plt_apps, :deps | :all_deps}` and we can default to `:all_deps` due to your report.